### PR TITLE
 Move plugin metainformation to a automatically generated manifest file; fixes #408

### LINKED ISSUE #408: Move plugin metainformation to a automatically generated manifest file
When building a plugin, an automatically generated manifest file with meta info about the plugin should be generated alongside the dll.

At very least should contain the info necessary for PluginManagerView.

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

- [x] manifest are auto generated
- [x] Packing
- [x] the plugin compilation can be done automatically

### CODE REVIEW

File: CDP4PluginPackager.Tests/AppTestFixture.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+using NUnit.Framework;
```
Comment (samatrhea) [inline CDP4PluginPackager.Tests/AppTestFixture.cs:1]: please add the appropriate header and make sure using statements are inside the namespace as per coding convention

File: CDP4PluginPackager/App.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="App.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Merlin Bieze, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru
```
Comment (samatrhea) [inline CDP4PluginPackager/App.cs:5]: please remove Mihail from authors

File: CDP4PluginPackager/App.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            this.Manifest = new Manifest
+            {
+                Name = this.AssemblyInfo.Name,
+                Version = this.AssemblyInfo.Version.ToString(),
+                ProjectGuid = this.Csproj.PropertyGroup.First(p => p.ProjectGuid != Guid.Empty).ProjectGuid,
+                TargetFramework = this.Csproj.PropertyGroup.First(p => !string.IsNullOrWhiteSpace(p.TargetFrameworkVersion)).TargetFrameworkVersion,
+                License = this.GetLicense(),
+                Author = "RHEA System S.A.",
+                References = this.ComputeReferences().ToList(),
+                Website = "https://store.cdp4.com",
```
Comment (samatrhea) [inline CDP4PluginPackager/App.cs:126]: store.cdp4.org

File: CDP4PluginPackager/CDP4PluginPackager.csproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
```
Comment (samatrhea) [inline CDP4PluginPackager/CDP4PluginPackager.csproj:4]: please add more tags related to
  - authors
  - company
  - copyright
  - license

etc.. see CDP4Services projects to get inspiration

File: CDP4PluginPackager/Models/AutoGen/Configuration.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4PluginPackager.Models.AutoGen
+{
+    using System.Xml.Serialization;
+
+    /// <summary>
+    /// Autogenerated class matching a "Configuration" tag in a csproj file
```
Comment (samatrhea) [inline CDP4PluginPackager/Models/AutoGen/Configuration.cs:32]: is this indeed autogenerated?

Comment (nathanatrhea) [reply CDP4PluginPackager/Models/AutoGen/Configuration.cs:32]: the classes were indeed generated, not the doc.

File: CDP4PluginPackager.Tests/AppTestFixture.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        
+        [Test]
+        public void VerifyIMEVersion()
+        {
+            var app = new App();
+            app.Deserialize();
+            app.GetAssemblyInfo();
+            var version = app.GetCurrentIMEVersion();
+            Assert.IsNotNull(version);
+            Assert.AreEqual(version.ToString(), "6.0.0.2");
```
Comment (nathanatrhea) [inline CDP4PluginPackager.Tests/AppTestFixture.cs:84]: change the hardcoded IME version

File: CDP4PluginPackager/Models/AutoGen/Target.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
+﻿// -------------------------------------------------------------------------------------------------
+// <copyright file="Target.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+// </copyright>
+// <summary>
+//   This is an auto-generated class. Any manual changes on this file will be overwritten!
+// </summary>
+// -------------------------------------------------------------------------------------------------
+
```
Comment (nathanatrhea) [inline CDP4PluginPackager/Models/AutoGen/Target.cs:9]: This file has been generated using a csproj file with this tool to convert Xml structure into Csharp : <link>

**Pull request merged**